crates/pallet-domains/src/lib.rs

```diff
@@ -298,6 +298,11 @@ mod pallet {
     #[pallet::storage]
     pub(super) type SuccessfulBundles<T> = StorageMap<_, Identity, DomainId, Vec<H256>, ValueQuery>;
 
+    /// Fraud proofs submitted successfully in current block.
+    #[pallet::storage]
+    pub(super) type SuccessfulFraudProofs<T: Config> =
```

Comment:
help me understand why we need to have this storage. How can a fraud proof is successfully validated at pre_dispatch and fail in call? Or is this used for something else?

Reply:
Generally `pre_dispatch` is called from transaction pool before call is dispatch, it may become invalid by the time it has to be included in the block.

Reply:
`SuccessfulFraudProofs` is used to serve the same purpose as `SuccessfulBundles`, it is not used in the runtime directly but in the runtime API to extract the successfully submitted fraud proof from the block body for the client, thus it is used to distinguish fraud proof that can pass `pre_dispatch` from those can't.